submit-api/src/Cardano/TxSubmit/Node.hs

```diff
@@ -146,78 +84,43 @@ newtype SocketPath = SocketPath
   }
 
 
-runTxSubmitNode :: TxSubmitVar -> Trace IO Text -> Genesis.Config -> SocketPath -> IO ()
-runTxSubmitNode tsv trce gc socket = do
+runTxSubmitNode
+  :: TxSubmitVar
+  -> Trace IO Text
+  -> Protocol
+  -> NetworkId
+  -> SocketPath
+  -> IO ()
+runTxSubmitNode tsv trce protocol nwId socket = do
   logInfo trce "Running tx-submit node"
-  logException trce "tx-submit-node." $ do
-    logProtocolMagic trce $ Ledger.configProtocolMagic gc
-    void $ runTxSubmitNodeClient tsv (mkNodeConfig gc) trce socket
+  logException trce "tx-submit-node." $
+    runTxSubmitNodeClient tsv trce protocol nwId socket
 
 
-mkNodeConfig :: Genesis.Config -> TopLevelConfig ByronBlock
-mkNodeConfig gc =
-  pInfoConfig . protocolInfo $ ProtocolRealPBFT gc Nothing (Update.ProtocolVersion 0 2 0)
-      (Update.SoftwareVersion (Update.ApplicationName "cardano-sl") 1) Nothing
-
 runTxSubmitNodeClient
   :: forall blk. (blk ~ ByronBlock)
-  => TxSubmitVar -> TopLevelConfig blk
-  -> Trace IO Text -> SocketPath
-  -> IO Void
-runTxSubmitNodeClient tsv nodeConfig trce (SocketPath socketPath) = do
-  logInfo trce $ "localInitiatorNetworkApplication: connecting to node via " <> textShow socketPath
-  networkState <- newNetworkMutableState
-  withIOManager $ \iocp -> do
-    ncSubscriptionWorker_V1
-      (localSnocket iocp socketPath)
-      -- TODO: these tracers should be configurable for debugging purposes.
-      NetworkSubscriptionTracers {
-          nsMuxTracer = nullTracer,
-          nsHandshakeTracer = nullTracer,
-          nsErrorPolicyTracer = errorPolicyTracer,
-          nsSubscriptionTracer = nullTracer
-          -- TODO subscription tracer should not be 'nullTracer' by default
-        }
-      networkState
-      ClientSubscriptionParams
-          { cspAddress = LocalAddress socketPath
-          , cspConnectionAttemptDelay = Nothing
-          , cspErrorPolicies = networkErrorPolicies <> (consensusErrorPolicy (Proxy @blk))
-          }
-      (NodeToClientVersionData
-        { networkMagic = nodeNetworkMagic (Proxy @blk) nodeConfig
-        }
-      )
-      (const $ localInitiatorNetworkApplication trce tsv)
+  => TxSubmitVar
+  -> Trace IO Text
+  -> Protocol
+  -> NetworkId
+  -> SocketPath
+  -> IO ()
+runTxSubmitNodeClient tsv trce protocol nwId (SocketPath socketPath) = do
+    logInfo trce $ "runTxSubmitNodeClient: connecting to node via " <> textShow socketPath
+    withlocalNodeConnectInfo protocol nwId socketPath $ \connectInfo ->
+      case localNodeConsensusMode connectInfo of
+        ByronMode{} -> do
+          (metrics, server) <- registerMetricsServer
+          connectToLocalNode connectInfo (localNodeClientProtocols metrics)
+          cancel server
```

Comment:
What's the best way to handle this? I'm not sure whether `connectToLocalNode` can throw an `IOException` or something.

Reply:
Can we try throw an `IOException` from here and see what it does?

Reply:
Actually, this might not be a big deal. In my upcoming PRs, we won't be using `connectToLocalNode` like this.